src/cli/pipeline-steps.ts

```diff
@@ -28,12 +28,59 @@ function extractFilesetFromDirectory(path: string) {
   return allEntries.filter(isFile).map(adaptToFileInterface(path));
 }
 
-export function extractFilesetFromPath(path: string) {
-  const fsInfo = fs.statSync(path);
+export function extractFilesetFromPath(pathStr: string) {
+  const fsInfo = fs.statSync(pathStr);
 
   if (fsInfo.isDirectory()) {
-    return extractFilesetFromDirectory(path);
+    return extractFilesetFromDirectory(pathStr);
   }
 
-  return []; // no other methods supported yet
+  if (fsInfo.isFile()) {
+    const sizeInBytes = fsInfo.size;
+    if (sizeInBytes > 50000) {
+      return [];
+    }
+    const text = fs.readFileSync(pathStr, "utf8");
+    let parsed;
+    try {
+      parsed = JSON.parse(text);
+    } catch (e) {
+      return [];
+    }
+    if (parsed.file_type === "OCF_MANIFEST_FILE") {
+      const dir = path.dirname(pathStr);
+      const referencedFiles = [];
+      for (const key of Object.keys(parsed)) {
+        if (key.endsWith("_files") && Array.isArray(parsed[key])) {
+          for (const ref of parsed[key]) {
+            if (ref && typeof ref.filepath === "string") {
+              referencedFiles.push(ref.filepath);
+            }
+          }
+        }
+      }
+      const manifestFileObj = {
+        path: pathStr,
+        isSameAs: (containerRelativePath: string) =>
+          path.resolve(pathStr) === path.resolve(containerRelativePath),
+        sizeInBytes: fs.statSync(pathStr).size,
+        readAsText: () => fs.readFileSync(pathStr, "utf8"),
+      };
+      const referencedFileObjs = referencedFiles.map((f) => {
+        const absPath = path.resolve(dir, f);
+        const relPath = path.relative(dir, absPath);
+        return {
+          path: relPath,
+          isSameAs: (containerRelativePath: string) =>
+            path.resolve(dir, relPath) ===
+            path.resolve(dir, containerRelativePath),
+          sizeInBytes: fs.statSync(absPath).size,
+          readAsText: () => fs.readFileSync(absPath, "utf8"),
+        };
+      });
```

Comment:
:question: Why are we duplicating so much of the logic of `adaptEntryToFileInterface` in these blocks? Yes there are differences but there is also a lot of duplication.